src/main/java/com/bgsoftware/superiorskyblock/commands/CmdWarp.java

```diff
@@ -8,69 +8,91 @@
-        return Collections.singletonList("warp");
-    }
-
-    @Override
-    public String getPermission() {
-        return "superior.island.warp";
-    }
-
-    @Override
-    public String getUsage(java.util.Locale locale) {
-        return "warp [" +
-                Locale.COMMAND_ARGUMENT_PLAYER_NAME.getMessage(locale) + "/" +
-                Locale.COMMAND_ARGUMENT_ISLAND_NAME.getMessage(locale) + "]";
-    }
-
-    @Override
-    public String getDescription(java.util.Locale locale) {
-        return Locale.COMMAND_DESCRIPTION_WARP.getMessage(locale);
-    }
-
-    @Override
-    public int getMinArgs() {
-        return 1;
-    }
-
-    @Override
-    public int getMaxArgs() {
-        return 2;
-    }
-
-    @Override
-    public boolean canBeExecutedByConsole() {
-        return false;
-    }
-
-    @Override
-    public void execute(SuperiorSkyblockPlugin plugin, CommandSender sender, String[] args) {
-        Pair<Island, SuperiorPlayer> arguments = args.length == 1 ? CommandArguments.getSenderIsland(plugin, sender) :
-                CommandArguments.getIsland(plugin, sender, args[1]);
-
-        Island island = arguments.getKey();
-
-        if(island == null)
-            return;
-
-        SuperiorPlayer superiorPlayer = plugin.getPlayers().getSuperiorPlayer(sender);
-        MenuWarpCategories.openInventory(superiorPlayer, null, island);
-    }
-
-    @Override
-    public List<String> tabComplete(SuperiorSkyblockPlugin plugin, CommandSender sender, String[] args) {
-        return args.length == 2 ? CommandTabCompletes.getPlayerIslandsExceptSender(plugin, sender, args[1],
-                plugin.getSettings().tabCompleteHideVanished) : new ArrayList<>();
-    }
+	@Override
+	public List<String> getAliases() {
+		return Collections.singletonList("warp");
+	}
+
+	@Override
+	public String getPermission() {
+		return "superior.island.warp";
+	}
+
+	@Override
+	public String getUsage(java.util.Locale locale) {
+		return "warp [" +
+				Locale.COMMAND_ARGUMENT_PLAYER_NAME.getMessage(locale) + "/" +
+				Locale.COMMAND_ARGUMENT_ISLAND_NAME.getMessage(locale) + "] [" +
+                Locale.COMMAND_ARGUMENT_ISLAND_WARP.getMessage(locale) + "]";
+	}
+
+	@Override
+	public String getDescription(java.util.Locale locale) {
+		return Locale.COMMAND_DESCRIPTION_WARP.getMessage(locale);
+	}
+
+	@Override
+	public int getMinArgs() {
+		return 1;
+	}
+
+	@Override
+	public int getMaxArgs() {
+		return 3;
+	}
+
+	@Override
+	public boolean canBeExecutedByConsole() {
+		return false;
+	}
+
+	@Override
+	public void execute(SuperiorSkyblockPlugin plugin, CommandSender sender, String[] args) {
+		Pair<Island, SuperiorPlayer> arguments = args.length == 1 ? CommandArguments.getSenderIsland(plugin, sender) :
+			CommandArguments.getIsland(plugin, sender, args[1]);
+
+		Island island = arguments.getKey();
+		if(island == null)
+			return;
+
+		SuperiorPlayer superiorPlayer = plugin.getPlayers().getSuperiorPlayer(sender);
+		if (args.length == 3)
```

Comment:
What if you run /is warp <warp-name>? It sounds like a valid command, but according to the code you must use /is warp <island> <warp-name>, which is not necessary.

Reply:
I'll go add a setting at config then I guess if they should prioritize the warp or the player.